### PR TITLE
Add support for named xcontent parsers to high level REST client

### PR DESCRIPTION
**Pull request by javanna**
NamedXContentRegistry will be used by the high level REST client to parse aggregation responses, and any section whose class type depends on a json key. There are currently on entries in the standard registry, but soon aggs and most likely suggesters will be added. Also it is possible for subclasses to provide additional named xcontent parsers.

### CODE REVIEW

**Comment by javanna** [comment]
maybe @nik9000 wants to have a look too.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientExtTests.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.mockito.Mockito.mock;
+
+public class RestHighLevelClientExtTests extends ESTestCase {
```
Comment (tlrx) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientExtTests.java]: Can we add a small content that describes the kind of tests? Or rename it to `RestHighLevelClientNamedXContentTests` maybe?

**Review by tlrx — APPROVED** [review]

**Review by cbuescher — APPROVED** [review]
LGTM

File: client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientExtTests.java (hunk 1 above)
Comment (javanna) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientExtTests.java]: sure will add a comment

**Pull request merged**